common/src/main/java/net/neoforged/gradle/common/extensions/subsystems/SubsystemsExtension.java

```diff
@@ -0,0 +1,29 @@
+package net.neoforged.gradle.common.extensions.subsystems;
+
+import net.minecraftforge.gdi.ConfigurableDSLElement;
+import net.neoforged.gradle.dsl.common.extensions.subsystems.Subsystems;
+import org.gradle.api.Project;
+import org.gradle.api.provider.ProviderFactory;
+
+import javax.inject.Inject;
+import java.util.Arrays;
+
+public abstract class SubsystemsExtension implements ConfigurableDSLElement<Subsystems>, Subsystems {
+    private final Project project;
+
+    @Inject
+    public SubsystemsExtension(Project project) {
+        this.project = project;
+
+        ProviderFactory providers = project.getProviders();
+        getDecompiler().getMaxMemory().convention(providers.gradleProperty("neogradle.subsystems.decompiler.maxMemory"));
```

Comment:
`neogradle.subsystems` should be a constant somewhere

Reply:
Create a utility class to get the properties for neogradle. Give it a method for the subsystem properties and that should deal with it.